src/video/gdk/SDL_gdktextinput.cpp

```diff
@@ -0,0 +1,345 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2023 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+/*
+  Screen keyboard and text input backend
+  for GDK platforms.
+*/
+#include "SDL_internal.h"
+#include "SDL_gdktextinput.h"
+
+#ifdef SDL_GDK_TEXTINPUT
+
+/* GDK headers are weird here */
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN
+#endif
+#include <Windows.h>
+#include <XUser.h>
+#include <XGameUI.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "../../events/SDL_keyboard_c.h"
+#include "../windows/SDL_windowsvideo.h"
+#include <SDL3/SDL_system.h>
+#include <SDL3/SDL_hints.h>
```

Comment:
This header is already included in SDL_internal.h. SDL_system.h might be as well.